app.py

```diff
@@ -83,7 +88,79 @@
 (NBEST, CONSTRAINTS, DECTREE, REATTACH, RELABEL, REPARSE, EDITDIST, TIME
 		) = range(8)
 # e.g., "NN-SB/Nom" => ('NN', '-SB', '/Nom')
+
+PUNCT_ESCAPING = [{"istring" : "(", "ptree_label": "LRB-p", "ptree_token": "-LRB-", "ctree_punct": "("},
+				  {"istring" : ")", "ptree_label": "RRB-p", "ptree_token": "-RRB-", "ctree_punct": ")"},
+				  {"istring" : "-", "ptree_label": "HYPH-p", "ptree_token": "-", "ctree_punct": "-"}]
+
+# punctuation labels based on the Penn Treebank tagset
+# [from nltk.help.upenn_tagset()]
+PUNCT_TAGS = { 
+	# opening parenthesis
+	"-LRB-": "LRB", "[" : "LRB", "{": "LRB",
+	# closing parenthesis
+	"-RRB-": "RRB", "]": "RRB", "}": "RRB",
+	# sentence terminator
+	".": ".", "!": ".", "?": ".",
+	# colon or ellipsis
+	":": ":", ";": ":", "...": ":",
+	# comma
+	",": ",",
+	# dash
+	"-": "HYPH", "--": "HYPH",
+	# opening quotation mark
+	"``": "``", "`": "``",
+	# closing quotation mark
+	"''": "''", "'": "''",
+	# dollar sign
+	"$": "$"
+}
+
+# pos tag for symbols (and symbol sequences) that don't have an idiosyncratic PTB tag (in PUNCT_TAGS)
+SYMBOL_TAG = "*"
+
+# default labels for 'ambiguous' symbols that can be punctuation or something else depending on context
+
+SYMBOL_DEFAULTS = {
+	"$": "N-Head",
+	"#": "N-Head",
+	"@": "P-Head",
+	"%": "N-Head",
+	"&": "Coordinator-Marker",
+	"-": "Coordinator-Marker",
+	"/": "Coordinator-Marker"
```

Comment:
TBC these defaults aren't actually used, right? It's an illustration of the likely correct label for the ambiguous symbols?

Reply:
Yes, that's right!

Reply:
OK I tweaked this.

Is there any value in having "$" as a tag given that the "$" token is usually a noun?